### PR TITLE
#105336 fix

### PR DESCRIPTION
**Pull request by sidewayss**
QTransform replaces QMatrix and a bit of code in updateState()

### CODE REVIEW

File: mscore/svggenerator.cpp [hunk 1]
```diff
@@ -185,7 +185,7 @@ class SvgPaintEnginePrivate : public QPaintEnginePrivate
 
     QBrush brush;
     QPen pen;
-    QMatrix matrix;
+//    QMatrix matrix; // OBSOLETE
```
Comment (Jojo-Schmitz) [inline mscore/svggenerator.cpp]: If it is obsolete, delete it.

File: mscore/svggenerator.cpp [hunk 2]
```diff
@@ -1126,10 +1129,28 @@ void SvgPaintEngine::updateState(const QPaintEngineState &state)
...
-    QMatrix mx = state.matrix();
-    _dx = mx.dx();
-    _dy = mx.dy();
+    // Translations, SVG transform="translate()", are handled separately from
+    // other transformations such as rotation. Qt translates everything, but
+    // other transformations do occur, and must be handled here.
+    QTransform t = state.transform();
+    if (t.m11() == t.m22()    // No scaling
+     && t.m12() == t.m21()) { // No rotation, etc.
+        // No transformation except translation
```
Comment (Jojo-Schmitz) [inline mscore/svggenerator.cpp]: style: we indent by 6 spaces, not by 4

Comment (Jojo-Schmitz) [inline mscore/svggenerator.cpp]: style: we indent by 6 spaces, not by 4, see https://musescore.org/en/developers-handbook/musescore-coding-rules#Indentation-%26-Whitespaces

**Comment by ZackTheCardshark** [comment]
This is the first time I'm aware of that there have been more than a hundred pull requests open at once.

File: mscore/svggenerator.cpp (hunk 2 above)
Comment (Jojo-Schmitz) [inline mscore/svggenerator.cpp]: style: we indent by 6 spaces, not by 4, see https://musescore.org/en/developers-handbook/musescore-coding-rules#Indentation-%26-Whitespaces, , the XML at the ebd of that page helps if you're using QT-Creator

**Comment by sidewayss** [comment]
OK, I've deleted the QMatrix, and I've checked that in.

As for the indentation, the entire module is indented at 4 spaces.  I'm just being consistent.  Do you really need me to indent this section differently than the rest of the file?  My understanding is that this is originally a Qt file, and so it came this way and stays this way.  This was not an issue for this file in my previous pull requests.

**Comment by Jojo-Schmitz** [comment]
So the wrong indentation went unnoticed until now, no more, no less.

**Comment by sidewayss** [comment]
OK, I've indented this section at 6 spaces.  If you want me to re-indent the whole file, that will take more time...
Also, fyi - I don't think it went unnoticed, I think there was a conscious decision to keep this file indented at 4 spaces for the sake of consistency.

**Comment by Jojo-Schmitz** [comment]
It fixes #105436 too, so should mention this in the commit message

**Comment by sidewayss** [comment]
I need to mention both issues in the commit message?  OK, that means I need to undo the commit and redo it? I don't see instructions for that in the Git Workflow.  I think you've told me before, but maybe you could provide instructions again for how to revert the commit.  I don't want to mess it up, and my command line git skill are still not advanced.  Thanks.

**Comment by Jojo-Schmitz** [comment]
'git commit --ammend',  change the message,  'git push -f'

**Comment by Jojo-Schmitz** [comment]
'amend'...

**Comment by sidewayss** [comment]
I'm trying various permutations of these options and it's not working.
So I figured out how to do it in separate commands, but I screwed it up the first time due to MuseScore issues starting with the # character, which is the comment character for this file in git.  As soon as it's done with this check I'll push it again.

**Comment by sidewayss** [comment]
what just happened?  I made no changes locally except to amend that message, and the first push -f worked just fine.

**Comment by sidewayss** [comment]
Well, it all worked out green in the end, but the previous travis build failed.
Hold on, it just started another travis build. And now that's green too. Good.  The failure must have been a fluke.  Let me know if it needs further amending.

File: mscore/file.cpp [hunk 3]
```diff
@@ -2540,76 +2540,80 @@ bool MuseScore::saveSvg(Score* score, const QString& saveName)
...
+                              paintElement(p, firstSL);
+                        }
+                  }
+            }
+            // 2nd pass: the rest of the elements
+            QList<const Element*> pel = page->elements();
+            qStableSort(pel.begin(), pel.end(), elementLessThan);
+
+            Element::Type eType;
+            foreach (const Element* e, pel) {
```
Comment (Jojo-Schmitz) [inline mscore/file.cpp]: while touching the code, you could as well change the all the `foreach (a, b)` into `for (a : b)`

File: mscore/file.cpp [hunk 4]
```diff
@@ -2540,76 +2540,80 @@ bool MuseScore::saveSvg(Score* score, const QString& saveName)
             p.translate(-r.topLeft());
 
       foreach (Page* page, score->pages()) {
```
Comment (Jojo-Schmitz) [inline mscore/file.cpp]: while touching the code, you could as well change the all the `foreach (a, b)` into `for (a : b)`

File: mscore/file.cpp [hunk 5]
```diff
@@ -2540,76 +2540,80 @@ bool MuseScore::saveSvg(Score* score, const QString& saveName)
...
-                }
-                if (byMeasure) { // Draw visible staff lines by measure
-                    for (MeasureBase* mb = s->firstMeasure(); mb != 0; mb = s->nextMeasure(mb)) {
-                        Measure* m = static_cast<Measure*>(mb);
-                        if (m->visible(i)) {
-                            StaffLines* sl = m->staffLines(i);
-                            printer.setElement(sl);
-                            paintElement(p, sl);
+            // 1st pass: StaffLines
+            foreach (System* s, *(page->systems())) {
```
Comment (Jojo-Schmitz) [inline mscore/file.cpp]: ditto

File: mscore/file.cpp (hunk 4 above)
Comment (sidewayss) [inline mscore/file.cpp]: OK.  I stole that code from elsewhere in MuseScore many moons ago.  So for (a : b) is more efficient?  I might adjust a couple of comments too, while I'm in there...

Comment (Jojo-Schmitz) [inline mscore/file.cpp]: `foreach(a, b)` is the Qt way, `for (a : b)` is the C++11 way and, as per https://musescore.org/en/developers-handbook/musescore-coding-rules#for, to be peferred.

**Comment by sidewayss** [comment]
I just pushed the foreach to for changes, including an additional instance of foreach in PaintElements().  There are other instances of foreach in the file, but they are in procedures I have never touched.  fyi - I stole the loops you flagged from navigator.cpp::paintEvent(), and if you search for "foreach (Page" you'll see 3 more files that use the same loop.

**Comment by Jojo-Schmitz** [comment]
Just replacing those that are in the code your currently working at is OK.

**Comment by sidewayss** [comment]
Well, I made one more change than that, but it's called by the code that changed, so...
I just realized that my branch is badly named - the 105336 should be 105346.  But apparently that doesn't matter.  I do feel that I should change it, but I have found that doing less is always better with git, so I'll leave it alone unless it bothers you too.

**Comment by sidewayss** [comment]
Do you mind one more commit?  I had forgotten to make your other change, variable cs to variable score in this line of code (line 2517 in file.cpp):
https://github.com/Jojo-Schmitz/MuseScore/commit/aa61b7356210c744cd6f6ba636308540616c7a84#diff-f8370d09c06388b91cf4f5a714e5344dR2517

I have it ready to commit if it doesn't screw up anything else.
And re: your comment on that page, I think it might make a difference when saveSVG() is called by other modules,which happens in fotomode.cpp, for example.  In my own code I've tried to always use the function argument, not the class variable, for precisely that reason.  In the case of the page format it might not make any difference, but as a general rule it's better to use the function arg "score" rather than the class variable "cs" in this situation.

**Comment by Jojo-Schmitz** [comment]
You'd have to rebase this anyway (due to the recent merge of the newlayout branch into master), while doing that you can easily sneak in more changes

**Comment by sidewayss** [comment]
Sorry, I need to clarify:  Do I need to rebase this branch and push it again, regardless of whether I commit this change?

**Comment by Jojo-Schmitz** [comment]
yes

**Comment by sidewayss** [comment]
Well, I updated my local master, but the rebase with this branch failed due to my last set of changes.  I do not have experience navigating the patch process, and so far I have not succeeded in resolving the conflict.  If you want to guide me, OK, but on my own I'll have to kill this PR, start a new branch, and remake the changes, which are minimal edits anyway.  Thanks.

**Comment by Jojo-Schmitz** [comment]
Ok, may better to start afresh then

**Comment by sidewayss** [comment]
I'll get the issue numbers right in the branch name this time :-)  Thankfully these are easily repeatable changes, surprising that the integration failed, I'm not exactly sure what it failed on.  I'll kill this PR and open a new one shortly.

**Comment by sidewayss** [comment]
I get a compile error here:
https://github.com/musescore/MuseScore/blob/master/libmscore/staff.cpp#L533
with the latest master.  The error is:
Z:\MuseScore\libmscore\staff.cpp:534: error: 'qInfo' was not declared in this scope
                      _barLineSpan, idx, score()->nstaves());
                                                           ^

**Comment by Jojo-Schmitz** [comment]
You need Qt 5.6. check the updated build instructions in the developers handbook

**Comment by Jojo-Schmitz** [comment]
See also http://dev-list.musescore.org/Towards-MuseScore-3-td7579817.html

**Comment by sidewayss** [comment]
Wow, so that happened in the past day or so.  OK, onward and upward.  Thanks for the info.
The screen captures on the developer's handbook need to be updated, they show 5.4 and 4.9.1 here:
https://musescore.org/en/developers-handbook/compilation/compile-instructions-windows-mingw-git#MinGW-4.9.2-and-Qt-5.6
I'm attempting to use the Qt Maintenance Tool, available in the Windows Control Panel as well, to upgrade to 5.6 from 5.4.  If that doesn't work then I'll start from scratch.  If it works and I might try to post some steps for upgrading instead of re-installing, since the online installer won't let you overwrite the existing C:\Qt directory.  So far it seems to be working...

**Comment by Jojo-Schmitz** [comment]
Maintenance tool is fine. Yes, the screenshots needs to get replaced, thanks for the reminder.

**Comment by lasconic** [comment]
@sidewayss if you need help setting up your build environment, don't hesitate to come say hello on IRC https://kiwiirc.com/client/irc.freenode.net/?nick=musescore|?#musescore

**Comment by sidewayss** [comment]
I tried connecting to IRC and it keeps timing out. With my build directory set to MuseScore/build.qtc, CMake fails very early with the message below.  It seems like it's looking to the 491 folder for mingw32-make.exe, but I can't see where to change that setting.  I have rest all my path variables and everything else, AFAIK.  Here's the message:

-- Check for working C compiler: C:/Qt/Tools/mingw492_32/bin/gcc.exe
CMake Error: Generator: execution of make failed. Make command was: "C:/Qt/Tools/mingw491_32/bin/mingw32-make.exe" "cmTC_50efc/fast"
-- Check for working C compiler: C:/Qt/Tools/mingw492_32/bin/gcc.exe -- broken
CMake Error at C:/Program Files (x86)/CMake/share/cmake-3.3/Modules/CMakeTestCCompiler.cmake:61 (message):
  The C compiler "C:/Qt/Tools/mingw492_32/bin/gcc.exe" is not able to compile
  a simple test program.

  It fails with the following output:

   Change Dir: Z:/MuseScore/build.qtc/CMakeFiles/CMakeTmp

**Comment by sidewayss** [comment]
I just submitted a new PR to replace this one: https://github.com/musescore/MuseScore/pull/2552

I'll leave this one open until the end of the day just in case it's useful as a reference.

**Comment by sidewayss** [comment]
PR #2552 has been integrated and replaces this one.  But I don't see how to delete the PR.  I can delete the entire branch, as it is obsolete.  Or does someone simply reject this PR now?  What is the correct action here?

**Comment by Jojo-Schmitz** [comment]
Just press the close button, mext to the comment one.

**Pull request closed without merging**